modules/formulize/templates/admin/ui.html

```diff
@@ -1,309 +1,339 @@
+  <{/if}>
+<{/if}>
+
+<{if $breadcrumbtrail}>
+<p id="admin-breadcrumbs">
+    <{counter start=0 assign=crumbcount print=false}>
+    <{foreach from=$breadcrumbtrail item=crumb}>
+        <{if $crumbcount}> &raquo; <{/if}>
+        <{if $crumb.url}>
+            <a href="ui.php?<{$crumb.url}>" class="breadcrumb-<{$crumbcount}>"><{$crumb.text}></a>
+            <{assign var='lasturl' value=$crumb.url}>
+            <{assign var='lasttext' value=$crumb.text}>
+        <{else}>
+            <{$crumb.text}>
+        <{/if}>
+    <{counter}>
+    <{/foreach}>
+</p>
+<{/if}>
+
+<!-- modified by Freeform Solutions, S.Gray, April 11, 2011 -->
+<{if $adminPage.needsave}>
+<div id="admin_toolbar">
+<div id=savebutton>
+<{if $adminPage.isSaveLocked}>
+	READ ONLY
+<{else}>
+    <input type="button" class="savebutton" id="save" value="<{$smarty.const._AM_HOME_SAVECHANGES}>"/>
+<{/if}>
+</div>
+<div id="savewarning" class="ui-corner-all"><{$smarty.const._AM_HOME_WARNING_UNSAVED}></div>
+<div id="derivedfinished" class="ui-corner-all" style="display:none"><{$smarty.const._AM_ELE_DERIVED_DONE}></div>
+</div><!-- /admin_toolbar -->
+<{/if}>
+
+<{if $adminPage.template}>
+<{include file=$adminPage.template}>
+<{/if}>
+
+<{if $adminPage.tabs}>
+<{include file="db:admin/ui-tabs.html" tabs=$adminPage.tabs}>
+<{/if}>
+
+<p class="versionnumber">Version <{$version}></p>
+
+</div><!-- End admin-ui -->
+
+<script type="text/javascript">
+
+	var icms_url = '<{$XOOPS_URL}>';
+	var icms_userid = <{$UID}>;
+  var saveCounter = 0;
+  var saveTarget = 0;
+  var redirect = "";
+  var newhandle = "";
+  var formdata = new Array();
+
+  $("input").change(function() {
+    setDisplay('savewarning','block');
+    });
+  $("input[type=text]").keydown(function() {
+    setDisplay('savewarning','block');
+    });
+  $("select").change(function() {
+        if ($(this).attr('name') != 'screens-theme') { // switching themes is a special event, and can only be done when there's been no changes to the settings yet
+            setDisplay('savewarning','block');
+        }
+    });
+  $("textarea").keydown(function() {
+    setDisplay('savewarning','block');
+    });
+
+  $(".savebutton").click(function() {
+		<{php}>
+		// if user wants it, and server can validate code, validate code blocks, which will then validateRequired, etc
+		global $xoopsModuleConfig;
+		if ($xoopsModuleConfig['validateCode'] AND isEnabled('shell_exec')) {
+			print "\t\t$('.admin-ui').fadeTo(1, 0.5, function() { validateCode(); });\n";
+		} else {
+			print "\t\tif(validateRequired()) { runSaveEvent(); }\n";
+		}
+		<{/php}>
+	});
+
+	async function validateCode() {
+		let codeBlocks = [];
+		let error = '';
+		let errorMsg = '';
+		$(".canValidate").each(function() {
+			codeBlocks.push({ code: $(this).val(), title: $(this).prevAll('legend').first().text() });
+		});
+		for (let i = 0; i < codeBlocks.length; i++) {
+			error = await fz_check_php_code(codeBlocks[i].code);
+			if(error.valid == false) {
```

Comment:
Suggestion: Usually best practice to perform a strict equality check.

```suggestion
			if(error.valid === false) {
```